### PR TITLE
add shapefile tests and refactor tests

### PR DESCRIPTION
**Pull request by ErickChacon**
This PR should be accepted after JuliaGeo/Shapefile.jl#90 is merged. It adds the following features:

- Conversion from and to `Shapefile.jl`, `GeoJSON,jl` and `ArchGDAL.jl` is working fine. So tests include `Shapefile.jl` conversions for `Chain` and `PolyArea`.
- Writing `GeoTables` with `Chain` and `PolyArea` geometries to `Shapefile.jl` is now posssible. So tests for `lines.*` and `polygons.*` to `.shp` are included.
- I have refactored the tests to reduce the number of lines.
- I also think would be better to write a the data to a temporal directory, so we do not need to remove those files.

### CODE REVIEW

**Comment by juliohm** [comment]
Yes, I realized later that we would need to remove the files manually if we save in the same data folder. Can you restore that temporary directory that you had in a previous PR? Maybe create it in the beginning of runtests.jl like we do with datadir and then use it everywhere.

**Comment by ErickChacon** [comment]
Done, I created a directory to write and then remove at the end.

File: test/runtests.jl [hunk 1]
```diff
@@ -12,6 +12,8 @@ import GeoJSON as GJS
 isCI = "CI" ∈ keys(ENV)
 islinux = Sys.islinux()
 datadir = joinpath(@__DIR__,"data")
+writedir = joinpath(tempdir(), "geotables")
+mkpath(writedir)
```
Comment (juliohm) [inline test/runtests.jl:16]:
```suggestion
writedir = tempdir()
mkpath(writedir)
```

**Review by juliohm — APPROVED** [review]

**Comment by juliohm** [comment]
@ErickChacon feel free to merge when it is appropriate. Also feel free to bump the path version and trigger the registration bot :)

File: test/runtests.jl [hunk 2]
```diff
@@ -309,3 +278,5 @@ datadir = joinpath(@__DIR__,"data")
     @test nitems(table) == 36
   end
 end
+
+rm(savedir, recursive = true)
```
Comment (juliohm) [inline test/runtests.jl:282]: I think we don't need to remove the temporary directory manually, the mktempdir function will do it upon exit.

**Comment by ErickChacon** [comment]
The PR in `Shapefile.jl` has been accepted however, the new version has not been registered. Should we remind it there?

**Comment by juliohm** [comment]
I think it is a good idea to remind them there of the missing patch release.

Em sáb., 6 de mai. de 2023 10:55, Dr. Erick A. Chacón Montalván <
***@***.***> escreveu:

**Comment by juliohm** [comment]
@ErickChacon the release was triggered for Shapefile.jl, can you try to update locally to see if tests are now passing as expected?

**Comment by ErickChacon** [comment]
All checks are passing locally. In our checks, there is only one test failing which is related to GADM.jl.

**Comment by ErickChacon** [comment]
@juliohm  It seems that I do not have write access to merge the PR?

**Comment by juliohm** [comment]
Awesome. Feel free to merge and release a new patch.

Em sáb., 6 de mai. de 2023 17:00, Dr. Erick A. Chacón Montalván <
***@***.***> escreveu:

**Comment by juliohm** [comment]
Sorry, I just added GeoTables.jl to the geostats team on GitHub that we are part of. You should be able to merge now, can you give a second try?

**Comment by ErickChacon** [comment]
> Sorry, I just added GeoTables.jl to the geostats team on GitHub that we are part of. You should be able to merge now, can you give a second try?

I do not have the option to merge the pull request, it says "Only those with write access ...."

**Comment by juliohm** [comment]
Ok, now it should work. The repository was in read mode in the team, now it is in write mode.

**Comment by ErickChacon** [comment]
I can see it now.

**Pull request merged**